### PR TITLE
Added cluster-clean command to clean all the resources

### PR DESCRIPTION
**Pull request by krsacme**
Eventhough cluster-deploy takes care of cleaning, it is useful to provide
an entry to clean the environment from makefile.

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically.
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

**Comment by rmohr** [comment]
@krsacme is that to clean up a cluster which is not exclusively for kubevirt testing?

**Comment by krsacme** [comment]
@rmohr yes and additionally it would be good to start fresh, like, while switching branches in kubevirt itself. And yes we could always do "vagrant destroy" and start fresh, but this could quicker than starting completely fresh.

**Comment by rmohr** [comment]
> @rmohr yes

that I understand

> and additionally it would be good to start fresh, like, while switching branches in kubevirt itself. And yes we could always do "vagrant destroy" and start fresh, but this could quicker than starting completely fresh.

that is exactly what make `cluster-sync` is for. We have the rule that every manifest we add needs the `kubevirt.io` label. This allows us to run `make cluster-sync` even when you switch branches. I use it like that all the time.

However, if you think an extra `make cluster-clean` helps you, I don't mind adding it. @davidvossel ?

File: Makefile [hunk 1]
```diff
@@ -59,7 +59,10 @@ cluster-down:
 cluster-build:
 	./cluster/build.sh
 
-cluster-deploy:
+cluster-clean:
```
Comment (stu-gott) [inline Makefile]: I think this also needs to be added to `.PHONY`.

**Comment by stu-gott** [comment]
Personally I think we should also have a `make cluster-destroy` Makefile target. It feels weird to need to invoke vagrant directly for that step.

**Comment by davidvossel** [comment]
> However, if you think an extra make cluster-clean helps you, I don't mind adding it. @davidvossel ?

I like what @krsacme has done here. I'd use cluster-clean before switching to a branch that contains manifests that aren't present in the other branch.  This has actually happened to me quite often.

File: Makefile (hunk 1 above)
Comment (krsacme) [reply Makefile]: ok, will add it.

**Comment by krsacme** [comment]
> Personally I think we should also have a make cluster-destroy Makefile target. It feels weird to need to invoke vagrant directly for that step.

I agree, I will work on it.

**Comment by rmohr** [comment]
> I like what @krsacme has done here. I'd use cluster-clean before switching to a branch that contains manifests that aren't present in the other branch. This has actually happened to me quite often.

When did that happen the last time? Before we added the release manifests? It is not the message I want to give people that they have to run `make cluster-clean` between switching branches. The labels on the manifest should be sufficient.

**Comment by rmohr** [comment]
ok to test

**Comment by davidvossel** [comment]
> When did that happen the last time? Before we added the release manifests? It is not the message I want to give people that they have to run make cluster-clean between switching branches. The labels on the manifest should be sufficient.

I have to manually clean things up any time I switch between branches that modify the manifests by adding/removing pods/deployments/etc...  Having 'make cluster-clean' means I can tear down one branch's manifests and then deploy using another branches manifests. Most changes people are working on won't require this. I'd like the option to have it though.

**Pull request merged**